docs/policy/authoring-guide.md

```diff
@@ -0,0 +1,970 @@
+
+    Use [Azure Built-In Roles table](https://docs.microsoft.com/azure/role-based-access-control/built-in-roles) to map the permission name to it's Resource ID.  Resource ID will be used when defining the role assignments. 
+
+4. Click on the **Duplicate initiative** button.  We will not be duplicating the policy set definition, but use this step to identify the parameter names that will need to be populated during policy assignment.
+
+    *Collect the following information from the **Initiative parameters** tab:*
+
+    * **Parameters** (i.e. `logAnalytics`, `logAnalyticsWorkspaceId`, `listOfResourceTypesToAuditDiagnosticSettings`).  You may see zero, one or many parameters listed.  It is possible that a policy set doesn't have any parameters.
+
+
+#### **Step 2: Create Bicep template & parameters JSON file**
+
+1. Navigate to `policy/builtin/assignments` directory and create two files.  Replace `POLICY_ASSIGNMENT` with the name of your assignment such as `pbmm`.
+
+   * POLICY_ASSIGNMENT.bicep (i.e. `pbmm.bicep`) - this file defines the policy assignment deployment
+   * POLICY_ASSIGNMENT.parameters.json (i.e. `pbmm.parameters.json`) - this file defines the parameters used to deploy the policy assignment.
+
+2. Edit the Bicep file to include the following template.  This template can be customized as required.  Pre-requisites are:
+
+    * targetScope must be `managementGroup`
+    * parameter `policyAssignmentManagementGroupId` must be defined.  It is used to set the policy assignment through automation.
+
+    **Sample Template**
+
+    ```bicep
+      targetScope = 'managementGroup'
+
+      @description('Management Group scope for the policy assignment.')
+      param policyAssignmentManagementGroupId string
+
+      // Start - Any custom parameters required for your policy assignment
+      param ...
+      // End - Any custom parameters required for your policy assignment
+
+      // Add the GUID from the Definition ID that was gathered above
+      var policyId = '<< GUID >>'
+
+      // Add the policy set assignment name (i.e. the name of the Policy Set Name)
+      var assignmentName = '<< POLICY ASSIGNMENT NAME >>'
+
+      var scope = tenantResourceId('Microsoft.Management/managementGroups', policyAssignmentManagementGroupId)
+      var policyScopedId = resourceId('Microsoft.Authorization/policySetDefinitions', policyId)
+
+      resource policySetAssignment 'Microsoft.Authorization/policyAssignments@2020-03-01' = {
+
+        // Set the name of the policy assignment
+        // Example: name: 'pbmm-${uniqueString('pbmm-',policyAssignmentManagementGroupId)}'
+
+        name: '<< NAME >>'
+
+        properties: {
+          displayName: assignmentName
+          policyDefinitionId: policyScopedId
+          scope: scope
+          notScopes: [
+          ]
+          parameters: {
+            // Add any parameters identified earlier into this section
+          }
+          enforcementMode: 'Default'
```

Comment:
Do we want to add comment describing allowable values for enforcement mode?

Reply:
Done